src/content/blog/astro-250.mdx

```diff
@@ -0,0 +1,167 @@
+
+**astro.config.mjs**
+
+```js
+import { defineConfig } from "astro/config"
+import onClickDirective from "astro-click-directive"
+
+export default defineConfig({
+	integrations: [onClickDirective()],
+	experimental: {
+		customClientDirectives: true,
+	},
+})
+```
+
+**astro-click-directive**
+
+```js
+export default function onClickDirective() {
+	return {
+		hooks: {
+			"astro:config:setup": ({ addClientDirective }) => {
+				addClientDirective({
+					name: "click",
+					entrypoint: "astro-click-directive/click.js",
+				})
+			},
+		},
+	}
+}
+```
+
+Now you can use `client:click` on any of your framework components with full type support.
+
+```astro
+<Counter client:click />
+```
+
+See the [LINK TO DOCS](#) to learn how to build these types of integrations.
+
+## HTML minification
+
+You can now opt-in to minification of the HTML produced by your Astro components.
+
+Using the `compressHTML` option Astro will remove all whitespace.
+
+We wanted to enable minification without sacrificing performance. For server-rendered apps, doing minification on each render could be costly. With `compressHTML` your components are compressed by the Astro compiler, only once and during the build.
+
+You can enable this in your config:
+
+```js
+import { defineConfig } from "astro/config"
+export default defineConfig({
+	compressHTML: true,
+})
+```
+
+Compression occurs both in development mode and in the final build.
+
+This option _will not_ compress HTML produced by framework components. If you'd like that functionality it can easily be achieved by writing a [middleware](https://docs.astro.build/en/guides/middleware/) to compress HTML responses.
+
+## Parallelized rendering
+
+When rendering components Astro will now render them in parallel, allowing components that do their own data loading to not be blocked by other components higher in the tree that are also loading data. A simple example:
+
+```astro
+<Delayed ms={30} />
+<Delayed ms={20} />
+<Delayed ms={40} />
+<Delayed ms={10} />
+```
+
+This `<Delayed />` component waits a number of milliseconds. In the past each component would need to wait for the previous to finish before it started rendering. In 2.5 these will now all render at the same time.
+
+We attempted to add this optimization in the past but were lacking good benchmarks to ensure that this didn't actually _slow down_ rendering. Recent improvements to our benchmarking CI infrastructure has now allowed us to implement this with confidence.
+
+## Polymorphic type helper
+
+Astro now includes a TypeScript helper (`Polymorphic`) to make it easier to build components that can render as different HTML elements with full type safety. This is useful for components like `<Link>` that can render as either `<a>` or `<button>` depending on the props passed to it.
+
+The example below implements a fully-typed, polymorphic component that can render as any HTML element. The HTMLTag type is used to ensure that the as prop is a valid HTML element.
```

Comment:
```suggestion
The example below implements a fully-typed, polymorphic component that can render as any HTML element. The HTMLTag type is used to ensure that the `as` prop is a valid HTML element.
```
I think this will be easier to read?